assets/js/base/context/hooks/cart/use-store-cart-event-listeners.ts

```diff
@@ -2,25 +2,50 @@
  * External dependencies
  */
 import { useEffect } from '@wordpress/element';
-import { CART_STORE_KEY as storeKey } from '@woocommerce/block-data';
+import { CART_STORE_KEY } from '@woocommerce/block-data';
 import { dispatch } from '@wordpress/data';
 import { translateJQueryEventToNative } from '@woocommerce/base-utils';
 
 interface StoreCartListenersType {
+	// Counts the number of consumers of this hook so we can remove listeners when no longer needed.
 	count: number;
+	// Function to remove all registered listeners.
 	remove: () => void;
 }
 
+interface CartDataCustomEvent extends Event {
+	detail?:
+		| {
+				preserveCartData?: boolean | undefined;
+		  }
+		| undefined;
+}
+
 declare global {
 	interface Window {
 		wcBlocksStoreCartListeners: StoreCartListenersType;
 	}
 }
 
-const refreshData = ( e ): void => {
-	const eventDetail = e.detail;
+const refreshData = ( event: CartDataCustomEvent ): void => {
+	const eventDetail = event?.detail;
 	if ( ! eventDetail || ! eventDetail.preserveCartData ) {
-		dispatch( storeKey ).invalidateResolutionForStore();
+		dispatch( CART_STORE_KEY ).invalidateResolutionForStore();
+	}
+};
+
+/**
+ * Refreshes data if the pageshow event is triggered by the browser history.
+ *
+ * The deprecated performance object is needed in chrome since persisted is not reliable.
+ */
+const refreshDataIfPersisted = ( event: PageTransitionEvent ): void => {
```

Comment:
Wondering if this function's name could be more descriptive of what it's doing? I never heard of "persisted" to mean loading from a cache before now, it makes sense, but I think it might be easier to reason with if we choose a simpler name. Could something like `refreshIfLoadedFromCache` or `refreshDataIfNavigationButtonsUsed` work better?